designer/client/outputs/output-edit.tsx

```diff
@@ -1,49 +1,75 @@
-import React from "react";
+import React, {
+  Component,
+  MouseEvent,
+  ChangeEvent,
+  FormEvent,
+  ReactNode,
+} from "react";
+import { clone } from "@xgovformbuilder/model";
+
 import NotifyEdit from "./notify-edit";
 import EmailEdit from "./email-edit";
-import { clone } from "@xgovformbuilder/model";
+import WebhookEdit from "./webhook-edit";
+
+import { OutPutType, OutputConfiguration, Output } from "./types";
 
-class OutputEdit extends React.Component {
-  constructor(props) {
+type State = {
+  outputType: OutPutType;
+};
+
+type Props = {
+  onEdit: ({ data: any }) => void; // TODO: type
+  onCancel: (event: MouseEvent<HTMLAnchorElement>) => void;
+  data: any; // TODO: type
+  output: Output;
+};
+
+class OutputEdit extends Component<Props, State> {
+  constructor(props: Props) {
     super(props);
-    this.state = { outputType: props.output?.type ?? "notify" };
+    this.state = { outputType: props.output?.type ?? OutPutType.Email };
   }
 
-  onSubmit = (e) => {
-    e.preventDefault();
+  onSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
     let output = this.props.output || { name: "", type: "" };
-    const form = e.target;
+    const form = event.currentTarget;
     const formData = new window.FormData(form);
     const { data } = this.props;
     const copy = clone(data);
-    const outputType = formData.get("output-type") || output.type;
-    const outputName = formData.get("output-name");
-    const outputTitle = formData.get("output-title");
-    let outputIndex;
+    const outputType: OutPutType =
+      (formData.get("output-type") as OutPutType) || output.type;
+    const outputName = formData.get("output-name") as string;
+    const outputTitle = formData.get("output-title") as string;
+
+    let outputIndex: number = -1;
+
     if (output.name) {
       outputIndex = data.outputs.indexOf(output);
     }
 
-    let outputConfiguration = output.outputConfiguration || {};
+    let outputConfiguration: OutputConfiguration =
+      output.outputConfiguration || {};
+
     switch (outputType) {
-      case "notify":
+      case OutPutType.Email:
```

Comment:
being pedantic, sorry! output is one word so `outputType`

Reply:
Not pedantic, wrong is wrong. Thanks